async_producer_test.go

```diff
@@ -1230,6 +1231,126 @@ func TestBrokerProducerShutdown(t *testing.T) {
 	mockBroker.Close()
 }
 
+type appendInterceptor struct {
+	i int
+}
+
+func (b *appendInterceptor) onSend(msg *ProducerMessage) {
+	if b.i < 0 {
+		panic("hey, the interceptor have failed")
+	}
+	v, _ := msg.Value.Encode()
+	msg.Value = StringEncoder(string(v) + strconv.Itoa(b.i))
+	b.i++
+}
+
+func (b *appendInterceptor) onConsume(msg *ConsumerMessage) {
+	if b.i < 0 {
+		panic("hey, the interceptor have failed")
```

Comment:
These panics should never be seen by any user, but either way `"the interceptor has failed"` feels more right to me 😄

Reply:
Thanks!